### PR TITLE
specify minimal dependency version

### PR DESCRIPTION
**Pull request by pawelru**
WIP

### CODE REVIEW

**Review by shajoezhu — APPROVED** [review]
lgtm! Thanks @pawelru

**Comment by shajoezhu** [comment]
Hi @pawelru , I wasn't sure why the vignette build is failing. I was wondering why is the minimal version failing this?

File: .github/workflows/integration-tests.yaml [hunk 1]
```diff
@@ -4,6 +4,10 @@ name: Integration Tests 🧪
 on:
   schedule:
     - cron: '45 3 * * 0'
+  workflow_dispatch:
+  push:
+    branches:
+      - min_deps
```
Comment (pawelru) [inline .github/workflows/integration-tests.yaml:10]: Need to revert it once completed
```suggestion
```

File: .github/workflows/integration-tests.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ name: Integration Tests 🧪
 on:
   schedule:
     - cron: '45 3 * * 0'
-  workflow_dispatch:
+
```
Comment (pawelru) [inline .github/workflows/integration-tests.yaml:7]:
```suggestion
  workflow_dispatch:
  push:
    branches:
      - min_deps
```

**Comment by averissimo** [comment]
Closed in favor of #59 that implements new strategies `min_isolated` and `min_cohort`

**Pull request closed without merging**